### PR TITLE
New build artefact cache location

### PR DESCRIPTION
**Pull request by sulkaharo**
When testing deploying the app to Digital Ocean's PAAS, I discovered the system caches the node_modules output and skips the build step on redeployment of the app, causing the application to fail on redeploy as this causes the webpack javascript bundles to be missing from the deployed application. This PR moves npm build output from /tmp to /node_modules/.cache folder, which is intended to be used for this exact purpose (see for example https://jongleberry.medium.com/speed-up-your-ci-and-dx-with-node-modules-cache-ac8df82b7bb0 ). This branch has been deployed to both Heroku and Digital Ocean and works as expected on both.

### CODE REVIEW

**Comment by bewest** [comment]
Do you want to get into `dev` now or release `dev` as-is and follow up with this and other things?  (I slightly prefer the latter.)

File: package.json [hunk 1]
```diff
@@ -35,7 +35,7 @@
     "bundle": "webpack --mode production --config webpack/webpack.config.js && npm run-script generate-keys",
     "bundle-dev": "webpack --mode development --config webpack/webpack.config.js && npm run-script generate-keys",
     "bundle-analyzer": "webpack --mode development --config webpack/webpack.config.js --profile --json > stats.json && webpack-bundle-analyzer stats.json",
-    "generate-keys": "node bin/generateRandomString.js >tmp/randomString",
+    "generate-keys": "mkdir node_modules/.cache/_ns_cache | node bin/generateRandomString.js >node_modules/.cache/_ns_cache/randomString",
```
[the remark this answers is not in the document]
Comment (mrinnetmaki) [reply package.json:38]: Of course, `mkdir` probably does not work cross platform anyway...

**Comment by mrinnetmaki** [comment]
@sulkaharo did you check the tests? They seem to fail because the build directory was chanced.

```1) admintools
[4282](https://github.com/nightscout/cgm-remote-monitor/runs/8067274317?check_suite_focus=true#step:7:4283)
       "before all" hook for "should produce some html":
[4283](https://github.com/nightscout/cgm-remote-monitor/runs/8067274317?check_suite_focus=true#step:7:4284)
     Uncaught Cannot find module '/home/runner/work/cgm-remote-monitor/cgm-remote-monitor/tests/../tmp/public/js/bundle.app.js'```

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (sulkaharo) [reply package.json]: Changed to &&. Maybe have this as a post script actually, the packaging process creates the directory, so mkdir could then be removed

[the remark this answers is not in the document]
Comment (sulkaharo) [reply package.json]: Removed mkdir entirely

**Comment by sulkaharo** [comment]
@bewest This is not critical for next release but would be great to get merged asap. I'm pretty sure DO is not the only cloud provider that breaks with the old packaging

**Pull request merged**